### PR TITLE
MNTOR-2711 - update database from subscription webhook before onerep

### PR DESCRIPTION
**Pull request by rhelmer**
# References: 
Jira: MNTOR-2711

# Description

When testing on staging, I encountered a situation where the subscription webhook came through but OneRep activation failed. In this case our webhook handler returns an HTTP error so FxA retries, but the badge doesn't show up right away. 

We should update our database first.

# How to test

I ran into this on stage, this can be tested by pointing the `ONEREP_API_BASE` var to an endpoint that always returns `500`.

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: src/app/api/v1/fxa-rp-events/route.ts [hunk 1]
```diff
@@ -286,8 +286,31 @@ export async function POST(request: NextRequest) {
             }
 
             // activate and opt out profiles
-            await activateProfile(oneRepProfileId);
-            await optoutProfile(oneRepProfileId);
+            try {
```
Comment (mansaj) [inline src/app/api/v1/fxa-rp-events/route.ts:289]: Not in this PR, but I'm wondering why we have two info logging for this case `FXA_SUBSCRIPTION_CHANGE_EVENT`, L244

File: src/app/api/v1/fxa-rp-events/route.ts [hunk 2]
```diff
@@ -245,45 +245,73 @@ export async function POST(request: NextRequest) {
...
-        });
+        try {
+          // get profile id
+          const result = await getOnerepProfileId(subscriber.id);
+          const oneRepProfileId = result?.[0]?.["onerep_profile_id"] as number;
 
-        // MNTOR-2103: if one rep profile id doesn't exist in the db, fail silently
-        if (!oneRepProfileId) {
-          logger.error("onerep_profile_not_found", {
+          logger.info("get_onerep_profile", {
```
Comment (mansaj) [inline src/app/api/v1/fxa-rp-events/route.ts:253]: already having a logger.info above?

File: src/app/api/v1/fxa-rp-events/route.ts [hunk 3]
```diff
@@ -245,45 +245,73 @@ export async function POST(request: NextRequest) {
           subscriber_id: subscriber.id,
```
Comment (mansaj) [inline src/app/api/v1/fxa-rp-events/route.ts:245]: already having a logger.info above (l244, l238)?

File: src/app/api/v1/fxa-rp-events/route.ts [hunk 4]
```diff
@@ -293,12 +321,35 @@ export async function POST(request: NextRequest) {
...
+                        )}`),
+              );
+              return NextResponse.json(
+                { success: false, message: "failed_activating_subscription" },
+                { status: 500 },
+              );
+            }
+
             // deactivation stops opt out process
             await deactivateProfile(oneRepProfileId);
```
Comment (mansaj) [inline src/app/api/v1/fxa-rp-events/route.ts:351]: Should we do the same with deactivate to catch for the 403?

**Review by mansaj — APPROVED** [review]
lgtm

File: src/app/api/v1/fxa-rp-events/route.ts (hunk 4 above)
Comment (rhelmer) [reply src/app/api/v1/fxa-rp-events/route.ts:351]: There isn't anything else to do after deactivation (there's no inverse of the "opt-out" API call), deactivate should also stop the opt-out process.

Comment (rhelmer) [reply src/app/api/v1/fxa-rp-events/route.ts:351]: Oh I see what you're saying though, if it's already de-activated. Yes I think it does make sense then.

File: src/app/api/v1/fxa-rp-events/route.ts (hunk 3 above)
Comment (rhelmer) [reply src/app/api/v1/fxa-rp-events/route.ts:245]: Yes this one confused me earlier, I think we just need the one that includes the payload.

**Pull request merged**